x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/termsenum/action/TransportTermsEnumAction.java

```diff
@@ -259,6 +259,9 @@ protected TermsEnumResponse mergeResponses(
                 successfulShards += rc.resp.getSuccessfulShards();
                 failedShards += rc.resp.getFailedShards();
                 for (DefaultShardOperationFailedException exc : rc.resp.getShardFailures()) {
+                    if (shardFailures == null) {
+                        shardFailures = new ArrayList<>();
```

Comment:
This fixes another related bug which was observed:

```
{
  "error": {
    "root_cause": [
      {
        "type": "null_pointer_exception",
        "reason": "Cannot invoke \"java.util.List.add(Object)\" because \"shardFailures\" is null"
      }
    ],
    "type": "null_pointer_exception",
    "reason": "Cannot invoke \"java.util.List.add(Object)\" because \"shardFailures\" is null"
  },
  "status": 500
}
```

Testing this one is tricky as you need a test with a remote cluster, and you need the remote cluster to fail the terms enum on some field for some reason. I've therefore opted to fix without test.